packages/flutter/lib/src/widgets/form.dart

```diff
@@ -293,19 +293,45 @@ class FormState extends State<Form> {
   /// returns true if there are no errors.
   ///
   /// The form will rebuild to report the results.
+  /// See also: [validateGranually]
   bool validate() {
     _hasInteractedByUser = true;
     _forceRebuild();
-    return _validate();
+    return _validate().isValid;
   }
 
-  bool _validate() {
+
+  /// Validates every [FormField]s that is a descendant of this [Form], and
+  /// returns a [Map] which keys' are [Key]s of the descendant [FormField]s
+  /// and values' are their corresponding validation results.
+  ///
+  /// Any [FormField] that has not been asigned a key will be ignored in the result.
+  /// To get the absolute validation value of the Form, Consider using [validate]
+  ///
+  /// Common usage of this method is when you need draw the user's attention
+  /// to the invalid field(s) using their widget key(s).
+  ///
+  /// The form will rebuild to report the results.
+  Map<Key, bool> validateGranually() {
```

Comment:
Maybe just return a collection of fields that `hasError`?

Reply:
Also the method name doesn't seem very self-explanatory when I first read it.

Reply:
> Maybe just return a collection of fields that hasError?

I'm avoiding returning the field itself because `FormFieldState` doesn't look like a class that's meant to be exposed to a public API. Otherwise, why `_fields` in `FormState` has no public getter in the first place? imo `FormFieldState` is a public class only to be used as a param to the [builder ](https://github.com/flutter/flutter/blob/58b69f34cb4451fc6d52c2d8a75d3e1072d4ab28/packages/flutter/lib/src/widgets/form.dart#L448)when extending `FormField`.

Please lmk if you disagree. In that case, we have two better approaches to achieving #135363 :
1. We can add a public getter for `_fields` and revert all the changes, as the user would be able to call [isValid ](https://github.com/flutter/flutter/blob/58b69f34cb4451fc6d52c2d8a75d3e1072d4ab28/packages/flutter/lib/src/widgets/form.dart#L523) on the fields individually.
2. We return a collection of all the fields with error as you're suggesting, thus resolving your other comment below regarding only including keyed fields.

Reply:
> Also the method name doesn't seem very self-explanatory when I first read it.

I totally agree but couldn't come up with a better name. Any suggestions?

Maybe:
- validateAndGetResults
- validateWithResults
- getInvalidFields (if we decided to return only the fields with error as you're suggesting)
- ..?
Optimally, I'd call this method `validate` and call the old method `validateAll`, but it's a breaking change so I don't think it's an option.

Reply:
`FormFieldState` is meant to be public I think (it has public methods, e.g., https://master-api.flutter.dev/flutter/widgets/FormFieldState/save.html). Also `Key`s aren't necessarily unique.

Reply:
@LongCatIsLooong I think it's the responsibility of the user to use unique keys for widgets that share a parent, otherwise they will get a "Duplicate Key found" error

Reply:
Yeah but `FormField`s don't have to share a parent. The `Form` just has to be the ancestor of all the fields.

Reply:
@LongCatIsLooong They will still get "Duplicate Key found" error in that case too. Using unique keys is the user's responsibility, even before this pr.

Reply:
While `GlobalKey`s must be unique in the widget tree, `Key`s don't have to be unique.